### DISCUSSION TITLE
Creating Android [MainApplication.cs] Missing Android Extension ShinyOnCreate(this Andro

### DISCUSSION THREAD

**Discussion by csharpcoder198** [shinyorg/shiny#1011] [Q&A] [answered]
Hello I am new to Shiny.

    I am busy adding Shiny to my Application based a lot on what I learned from Docs and especially the GeoFencing Sample.

// Head Project
I have succeeded at getting my Head project compiled and ready. 
1. Included required Shiny 2.51 Pkgs [Shiny, Shiny.Locations, Shiny.Notifications]
2. Created a class called Startup : ShinyStartup in Head project

Now I am turning to my existing Android Application

// Android Project
1. Included required Shiny 2.51 Pkgs [Shiny, Shiny.Locations, Shiny.Notifications] in the Android project
2. Created a MainApplication.cs see image below.
4. It is not pickling up the Android Extension Classes inside Shiny. Shiny.OnCreate(....) I wonder Why?
![image](https://user-images.githubusercontent.com/105575743/168884442-97f5bca6-7340-48c7-a83e-a68be492ef41.png)

**Comment by aritchie** [comment, marked as answer]
Your major target isn't Android 12.  Please be sure to read the nuget targets and release notes for a particular version.

**Comment by csharpcoder198** [comment]
okay i know what you mean about ReleeaseNotes and you were correct i was still on 11.0 I changed and got new error. I

what does it mean read nuget.targets? I have 2.5.1 on Shiny. *.Notifications *.Locations. But that is not what you mean.

My new error after switching  to TargetFramework 12 was:
1.  Xamarin.Android.Common.targets(1432, 3): [XA4210] Please add a reference to Mono.Android.Export.dll when using ExportAttribute or ExportFieldAttribute. 
2.    "/Users/markwardell/Documents/src/GA360/GA360.Android/GA360.Android.csproj" (build target) (2) ->
       (_GenerateJavaStubs target) ->      /Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(1432,3): error XA4210: Please add a reference to Mono.Android.Export.dll when using ExportAttribute or ExportFieldAttribute.

**Reply by csharpcoder198** [reply]
Solved it:

In my Android.csproj I needed to Add the Export -> yes just like the GeoFence Sample :(
  <Reference Include="Mono.Android.Export" />
